src/shared/sam/cli/samCliLocator.ts

```diff
@@ -69,8 +72,37 @@ abstract class BaseSamCliLocator {
 
         for (const fullPath of fullPaths) {
             this.logger.verbose(`Searching for SAM CLI in: ${fullPath}`)
+            const context: SamCliValidatorContext = {
+                samCliLocation: async () => fullPath,
+                getSamCliExecutableId: async () => 'temp',
+                getSamCliInfo: async () => {
+                    const samCliInfo = new SamCliInfoInvocation(
+                        new DefaultSamCliProcessInvoker({
+                            cliConfig: {
+                                getOrDetectSamCli: async () => {
+                                    return { path: fullPath, autoDetected: false }
+                                },
+                            },
+                        })
+                    )
+
+                    return await samCliInfo.execute()
+                },
+            }
+            const validator = new DefaultSamCliValidator(context)
             if (await filesystemUtilities.fileExists(fullPath)) {
-                return fullPath
+                try {
+                    const validationResult = await validator.getVersionValidatorResult()
+                    if (validationResult.validation === SamCliVersionValidation.Valid) {
+                        return fullPath
+                    } else {
+                        this.logger.info(
+                            `Detected SAM executable at ${fullPath} invalid: ${validationResult.validation}`
```

Comment:
This is going to be hard to read in the logs. The path is longer than "valid" so the path should be at the end. Not the middle of the log message.
```suggestion
                            `Found invalid SAM executable (${validationResult.validation}): ${fullPath}`
```